server/services/gdrive/gdrive.service.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Service for managing all google drive api workflow specifically generating
+ * snapshots or prevote documents.
+ *
+ * {
+ *   document_type: 'prevote' or 'snapshot' or 'both'
+ *   company_id:
+ * }
+ *
+ * See: http://isd-soft.com/tech_blog/accessing-google-apis-using-service-account-node-js/
+ */
+import errors from '@feathersjs/errors';
+import config from 'config';
+import { google } from 'googleapis';
+import App from '../../../client/schemas/app';
+import { DocumentTypes } from '../../../client/schemas/gdrive';
+import hooks from './gdrive.hooks';
+
+class GDriveService {
+  private app!: App;
+
+  setup(app) {
+    this.app = app;
+  }
+
+  async create(data) {
+    /*
+     * Get the documents we need to create
+     */
+    let toCreate;
+    if (data.document_type === DocumentTypes.Both) {
```

Comment:
Can we do this by using two API calls instead of the Both type? I think the Both type is unnecessary complexity.

Reply:
I was thinking this but that would have more chance of failure. Don't think this service is that complex.